File: api-angular/StatinaryProject/Model/Product_Information.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace StatinaryProject.Model
{
    public class Product_Information
    {
        [Key]
        public int Product_Id { get; set; }

      
        public string Product_Name { get; set; }

        public string Quantity { get; set; }

       
        public int Price { get; set; }

        public bool IsAvailable { get; set; }

        public DateTime OrderDate { get; set; }

        public string Image { get; set; }

        [NotMapped]
        public string? Shop_Name { get; set; }

        public int Shop_Id { get; set; }
        [ForeignKey("Shop_Id")]

        public virtual Shop_Information? Shop_Information { get; set; }

    }
}

File: api-angular/Stationary_AngularF/StationaryProjectF/src/app/Model/Product_Information.ts

export class Product_Information 
{

    Product_Id? : number;
    Product_Name? : string;
    Quantity? : string;
    Price? :number;
    IsAvailable? : Boolean;
    OrderDate? : Date;
    Image? : string;
    Shop_Id? : number;
    Shop_Name? : string;


    constructor(_Product_Id? : number, _Product_Name?:string, _Quantity?: string, _Price?:number,
         _IsAvailable?:Boolean, _OrderDate?:Date, _Image? : string, _Shop_Id? : number, _Shop_Name? : string  )
         {

    this.Product_Id= _Product_Id;
    this.Product_Name= _Product_Name;
    this.Quantity=_Quantity;
    this.Price=_Price;
    this.IsAvailable=_IsAvailable;
    this.OrderDate=_OrderDate;
    this.Image=_Image;
    this.Shop_Id=_Shop_Id;
    this.Shop_Name=_Shop_Name;

         }
}
